pkgs/dart_mcp/lib/src/api.dart

```diff
@@ -646,110 +654,151 @@ extension type ResourceTemplate.fromMap(Map<String, Object?> _value) {
-//      */
-//     name: string;
-//     /**
-//      * Arguments to use for templating the prompt.
-//      */
-//     arguments?: { [key: string]: string };
-//   };
-// }
+/// The server's response to a prompts/list request from the client.
+extension type ListPromptsResult.fromMap(Map<String, Object?> _value)
+    implements PaginatedResult {
+  factory ListPromptsResult({
+    required List<Prompt> prompts,
+    Cursor? cursor,
+    Meta? meta,
+  }) => ListPromptsResult.fromMap({
+    'prompts': prompts,
+    if (cursor != null) 'cursor': cursor,
+    if (meta != null) '_meta': meta,
+  });
 
-// /**
-//  * The server's response to a prompts/get request from the client.
-//  */
-// export interface GetPromptResult extends Result {
-//   /**
-//    * An optional description for the prompt.
-//    */
-//   description?: string;
-//   messages: PromptMessage[];
-// }
+  List<Prompt> get prompts => (_value['prompts'] as List).cast<Prompt>();
+}
 
-// /**
-//  * A prompt or prompt template that the server offers.
-//  */
-// export interface Prompt {
-//   /**
-//    * The name of the prompt or prompt template.
-//    */
-//   name: string;
-//   /**
-//    * An optional description of what this prompt provides
-//    */
-//   description?: string;
-//   /**
-//    * A list of arguments to use for templating the prompt.
-//    */
-//   arguments?: PromptArgument[];
-// }
+/// Used by the client to get a prompt provided by the server.
+extension type GetPromptRequest.fromMap(Map<String, Object?> _value)
+    implements Request {
+  static const methodName = 'prompts/get';
 
-// /**
-//  * Describes an argument that a prompt can accept.
-//  */
-// export interface PromptArgument {
-//   /**
-//    * The name of the argument.
-//    */
-//   name: string;
-//   /**
-//    * A human-readable description of the argument.
-//    */
-//   description?: string;
-//   /**
-//    * Whether this argument must be provided.
-//    */
-//   required?: boolean;
-// }
+  factory GetPromptRequest({
+    required String name,
+    Map<String, Object?>? arguments,
+    Meta? meta,
+  }) => GetPromptRequest.fromMap({
+    'name': name,
+    if (arguments != null) 'arguments': arguments,
+    if (meta != null) '_meta': meta,
+  });
 
-// /**
-//  * The sender or recipient of messages and data in a conversation.
-//  */
-// export type Role = "user" | "assistant";
+  /// The name of the prompt or prompt template.
+  String get name => _value['name'] as String;
 
-// /**
-//  * Describes a message returned as part of a prompt.
-//  *
-//  * This is similar to `SamplingMessage`, but also supports the embedding of
-//  * resources from the MCP server.
-//  */
-// export interface PromptMessage {
-//   role: Role;
-//   content: TextContent | ImageContent | EmbeddedResource;
-// }
+  /// Arguments to use for templating the prompt.
+  Map<String, Object?>? get arguments =>
+      (_value['arguments'] as Map?)?.cast<String, Object?>();
```

Comment:
nit: a cleanup throughout this API could be to use static consts for these repeated raw strings.

Reply:
Noted. Raw strings sgtm then.